feat: add privacy summary article (#554)
ipfs/ipfs-docs: docs/concepts/privacy.md

---
title: Privacy
sidebarDepth: 0

> description: Learn about user privacy and IPFS, the InterPlanetary File System.

Reviewer: ```suggestion
description: Learn about user privacy in IPFS, and why it does not come with a built-in privacy layer or encryption.
```

related:
  'Article: IPFS Privacy (Textile)': https://medium.com/pinata/ipfs-privacy-711f4b72b2ea
---

# IPFS and privacy

> As you explore IPFS and learn about its core concepts, it's important to know that as a protocol for peer-to-peer data storage and delivery, IPFS in itself does not explicitly have a "privacy layer" built in. This is in line with key principles of protocol design, since different uses of IPFS over its (hopefully very long) life may call for different approaches to privacy. Explicitly implementing an approach to privacy within the IPFS core could "box in" future builders and implementers due to a lack of modularity, flexibility, and future-proofing.

Reviewer: ```suggestion
As you explore IPFS and learn about its core concepts, it's important to know that as a protocol for peer-to-peer data storage and delivery, IPFS in itself does not explicitly have a "privacy layer" built in. This is in line with key principles of protocol design, since different uses of IPFS over its life may call for different approaches to privacy. Explicitly implementing an approach to privacy within the IPFS core could "box in" future builders and implementers due to a lack of modularity, flexibility, and future-proofing.
```

> By design, IPFS is a public network: Nodes participating in the network store data affiliated with globally consistent [content addresses](/concepts/content-addressing) (CIDs), and advertise that they have those CIDs available for other nodes to use via publicly viewable [distributed hash tables](/concepts/dht/) (DHTs). Indeed, this is one of IPFS's core strengths — at its most basic, it's a sort of globally distributed "server" of the network's total available data, referenceable both by the content itself (those CIDs) and by the participants who have or want the content (the nodes).

Reviewer: ```suggestion
By design, IPFS is a public network: Nodes participating in the network store data affiliated with globally consistent [content addresses](/concepts/content-addressing) (CIDs), and advertise that they have those CIDs available for other nodes to use through publicly viewable [distributed hash tables](/concepts/dht/) (DHTs). Indeed, this is one of IPFS's core strengths — at its most basic, it's a sort of globally distributed "server" of the network's total available data, referenceable both by the content itself (those CIDs) and by the participants who have or want the content (the nodes).
```

> What this does mean, however, is that IPFS itself isn't explicitly protecting knowledge _about_ CIDs and the nodes that provide or retrieve them. This isn't something unique to the distributed web; on both the d-web and the legacy web, traffic, and other metadata can be monitored in ways that can infer a lot about a network and its users. Some key details on this are outlined below, but in summary: IPFS traffic, including which nodes are retrieving and/or reproviding which CIDs, occurs in the clear. If you're worried about the implications of this for your own personal use case, it's worth taking additional measures such as encrypting sensitive content, accessing IPFS content [through Tor gateways](https://dweb-primer.ipfs.io/avenues-for-access/tor-gateways), or even running a private IPFS network if that's appropriate for you.

Reviewer 2: > **IPFS traffic,** including which nodes are retrieving and/or reproviding which CIDs, **occurs in the clear**

I wonder if  we need to clarify this in the sentence above and the `::: tip` below.

 I am aware this is a mental shortcut for non-technical reader to highlight that activity may be monitored by other peers, but "in the clear" part will confuse technical ones a lot: they either think "no encryption? wtf!" or know the traffic between peers is encrypted, so not "in the clear", and get bit confused. 

Not sure if we can have cookie and eat cookie, but i'd like to highlight that (1) p2p traffic is encrypted (2) (meta)data is public because anyone can participate in p2p exchanges

How about:

> IPFS traffic between peers is encrypted, however metadata that is published to DHT is public. When IPFS node participates in the public network, it announces CID of data in its cache, own PeerID and addresses on DHT and due to this other peers can learn which nodes are retrieving and/or reproviding which CIDs.

Reviewer: How about this:

> IPFS traffic is encrypted. However, when an IPFS node interacts with the public network it announces the CIDs it has stored. This makes it possible for other peers to learn which nodes are providing and requesting which CIDs.

Author: I believe https://github.com/ipfs/ipfs-docs/pull/554/commits/0a1e8565f8582d811ce12c33f6e1f0615f727e50 takes care of this, but would appreciate one more look from @lidel if possible.

Also reused the term "PeerID" in the rest of the guide, since I realize I wasn't explicit enough about that term of art.

::: tip
IPFS traffic, including which nodes are retrieving and/or reproviding what data, occurs in public. If you're worried about the implications of this for your personal use case, it's worth taking additional measures.
:::

## What's public on IPFS

All traffic on IPFS is public, including the contents of files themselves (unless they're encrypted; more about this below). For purposes of understanding IPFS privacy, this may be easiest to think about in two halves: content identifiers (CIDs), and IPFS nodes themselves.

### Content identifiers

> Because IPFS uses [content addressing](/concepts/content-addressing/) rather than the legacy web's method of location addressing, each piece of data stored in the IPFS network gets its own unique content identifier (CID). Copies of the data associated with that CID can be stored in any number of locations worldwide on any number of participating IPFS nodes. To make retrieving the data associated with a particular CID efficient and fast, IPFS uses a [distributed hash table](http://localhost:8080/concepts/dht/) (DHT) to keep track of what's stored where. When you use IPFS to retrieve a particular CID, your node queries the DHT to find the closest nodes to you with that item — and by default also agrees to reprovide that CID to other nodes for a limited time, until periodic "garbage collection" clears your cache of content you haven't used in a while. (You can also "pin" CIDs that you want to make sure are never garbage-collected, which also means you're acting as a permanent reprovider of that data.)

Reviewer: ```suggestion
Because IPFS uses [content addressing](/concepts/content-addressing/) rather than the legacy web's method of location addressing, each piece of data stored in the IPFS network gets its own unique content identifier (CID). Copies of the data associated with that CID can be stored in any number of locations worldwide on any number of participating IPFS nodes. To make retrieving the data associated with a particular CID efficient and fast, IPFS uses a [distributed hash table](http://localhost:8080/concepts/dht/) (DHT) to keep track of what's stored where. When you use IPFS to retrieve a particular CID, your node queries the DHT to find the closest nodes to you with that item — and by default also agrees to reprovide that CID to other nodes for a limited time, until periodic "garbage collection" clears your cache of content you haven't used in a while. You can also "pin" CIDs that you want to make sure are never garbage-collected, which also means you're acting as a permanent reprovider of that data.
```

> Because IPFS uses [content addressing](/concepts/content-addressing/) rather than the legacy web's method of location addressing, each piece of data stored in the IPFS network gets its own unique content identifier (CID). Copies of the data associated with that CID can be stored in any number of locations worldwide on any number of participating IPFS nodes. To make retrieving the data associated with a particular CID efficient and fast, IPFS uses a [distributed hash table](http://localhost:8080/concepts/dht/) (DHT) to keep track of what's stored where. When you use IPFS to retrieve a particular CID, your node queries the DHT to find the closest nodes to you with that item — and by default also agrees to reprovide that CID to other nodes for a limited time, until periodic "garbage collection" clears your cache of content you haven't used in a while. You can also "pin" CIDs that you want to make sure are never garbage-collected, which also means you're acting as a permanent reprovider of that data.

Reviewer 2: ```suggestion
Because IPFS uses [content addressing](/concepts/content-addressing/) rather than the legacy web's method of location addressing, each piece of data stored in the IPFS network gets its own unique content identifier (CID). Copies of the data associated with that CID can be stored in any number of locations worldwide on any number of participating IPFS nodes. To make retrieving the data associated with a particular CID efficient and fast, IPFS uses a [distributed hash table](/concepts/dht/) (DHT) to keep track of what's stored where. When you use IPFS to retrieve a particular CID, your node queries the DHT to find the closest nodes to you with that item — and by default also agrees to reprovide that CID to other nodes for a limited time, until periodic "garbage collection" clears your cache of content you haven't used in a while. You can also "pin" CIDs that you want to make sure are never garbage-collected, which also means you're acting as a permanent reprovider of that data.
```

Author: Hahaha ... thanks for spotting that one 😊

### Node IDs and node requests

The other half of the equation when considering the prospect of IPFS traffic monitoring is that nodes' unique identifiers are themselves public. Just like with CIDs, every individual IPFS node has its own public identifier, such as `QmRGgYP1P5bjgapLaShMVhGMSwGN9SfYG3CM2TfhpJ3igE`.

While a long string of letters and numbers may not be a "Johnny Appleseed" level of human-readable specificity, it's still a long-lived, unique identifier for your node. Keep in mind that it's possible to do a DHT lookup on your node's identifier and, particularly if your node is regularly running from the same location (like your home), find your IP address. Additionally, longer-term monitoring of the public IPFS network could yield information about what CIDs your node is requesting and/or reproviding and when.

## Enhancing your IPFS privacy

If there are situations in which you know you'll need to remain private but still want to use IPFS, one of the approaches outlined below may help. And don't forget, you can always discuss privacy and get others' input or ideas in the official [IPFS forums](https://discuss.ipfs.io).

> 

Reviewer 2: Useful to mention that one can tweak default provider setting of their node:

```suggestion

### Disable reproviding

By default IPFS node will announce itself as a provider for every CID in its cache. One can change that behavior by adjusting [reprovider settings](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#reprovider) in node's config.

Be aware that other peers won't be able to find the content if nobody is providing it, so make sure you pin and provide content that you care about.

```

Author: Excellent point. Added this in https://github.com/ipfs/ipfs-docs/pull/554/commits/1cd6c4ddbfc5154d5beeb09490e9843a20c478c2.

### Using a public gateway

> Using a public [IPFS gateway](https://dweb-primer.ipfs.io/avenues-for-access/review-these-lessons-from-the-tutorial-on-interacting-with-the-classical-http-web/public-gateways) is one way to request IPFS-hosted content without revealing any information about your local node — because you aren't using a local node! However, this method does keep you from enjoying all the benefits of being a full participant in the IPFS network.

Reviewer 2: Would internal link be better? 
We have bit more up to date section related to http gateways that includes basic intro to subdomain ones:

```suggestion
Using a public [IPFS gateway](/how-to/address-ipfs-on-web/#http-gateways) is one way to request IPFS-hosted content without revealing any information about your local node — because you aren't using a local node! However, this method does keep you from enjoying all the benefits of being a full participant in the IPFS network.
```

Author: Was going back and forth on this one -- I like that the dweb primer example is novice-friendly, but the docs example does mention subdomain gateways, which is important. We're also providing a handful of other dweb primer links, which does provide a path for the more novice readers. Good call.

Public IPFS gateways are primarily intended as a "bridge" between the legacy web and the distributed web; they allow ordinary web clients to request IPFS-hosted content via HTTP. That's great for back-compatibility, but if you only request content through public gateways rather than directly over IPFS, you're not actually part of the IPFS network; that gateway is the network participant, acting on your behalf. It's also important to remember that gateway operators could be collecting their own private metrics, which could include tracking the IP addresses that use a gateway and correlating those with what CIDs are requested.

### Using Tor

> If you're familiar with [Tor](https://www.torproject.org/), using the Tor browser and an IPFS-Tor gateway is a way to retrieve content from IPFS directly over Tor protocol. You can learn more about this experimental method in the [IPFS Primer](https://dweb-primer.ipfs.io/avenues-for-access/tor-gateways). If you're more adventurous and comfortable with the command line, you may also wish to try [running IPFS over Tor transport](https://dweb-primer.ipfs.io/avenues-for-access/tor-transport) by configuring your node's settings.

Reviewer 2: Flag: dweb primer article is bit outdated, which creates issues:

* `ipfs4uvgthshqonk.onion` I don't believe we are running public gateweay on tor
  * https://ipfs4uvgthshqonk.onion/ipns/en.wikipedia-on-ipfs.org does not load

Author: Drat - removed material about the gateway for safety's sake in https://github.com/ipfs/ipfs-docs/pull/554/commits/21e0e929e925833acc85d50f848ef7bc41878df7.

If you're a developer building on IPFS, it's worth noting that the global IPFS community continues to experiment with using Tor transport — see [this example from e-commerce organization OpenBazaar](https://github.com/OpenBazaar/go-onion-transport) — and there may already be an open-source codebase to help your own project achieve this.

### Encrypting content transported via IPFS

If your privacy concerns are less about the potential for monitoring and more about the visibility of the IPFS-provided content itself, this can be mitigated simply by encrypting the content before adding it to the IPFS network. While traffic involving the encrypted content could still be tracked, the _data_ represented by encrypted content's CIDs remains unreadable by anyone without the ability to decrypt it.

If you're curious about implementing encryption with IPFS on a large scale, you may enjoy reading [this case study on Fleek, a fast-growing IPFS file hosting and delivery service](concepts/case-study-fleek/).

### Creating a private network

> Private IPFS networks provide full protection from public monitoring but can lack the scale benefits provided by the public IPFS network. A private network operates identically to the public one, but with one critical difference: it can only be accessed by nodes that have been given access, and it will only ever scale to those nodes. This means that the benefits of the public IPFS network's massive scale, such as geographic resiliency and speedy retrieval of high-demand content, won't be realized unless the private network is explicitly designed and scaled with this in mind.

Reviewer 2: Would be good to link somewhere, sadly all we have are "experimental" config docs from go-ipfs (better than nothing?)

```suggestion
[Private IPFS networks](https://github.com/ipfs/go-ipfs/blob/v0.7.0/docs/experimental-features.md#private-networks) provide full protection from public monitoring but can lack the scale benefits provided by the public IPFS network. A private network operates identically to the public one, but with one critical difference: it can only be accessed by nodes that have been given access, and it will only ever scale to those nodes. This means that the benefits of the public IPFS network's massive scale, such as geographic resiliency and speedy retrieval of high-demand content, won't be realized unless the private network is explicitly designed and scaled with this in mind.
```

Author: Thanks for finding that - I knew there was some basic guidance out there but was struggling to find.

---
title: Privacy
sidebarDepth: 0
description: Learn about user privacy in IPFS and why it does not come with a built-in privacy layer or encryption.
related:
  'Article: IPFS Privacy (Textile)': https://medium.com/pinata/ipfs-privacy-711f4b72b2ea
---

# IPFS and privacy

> As you explore IPFS and learn about its core concepts, it's important to know that as a protocol for peer-to-peer data storage and delivery, IPFS in itself does not explicitly have a "privacy layer" built-in. This is in line with key principles of protocol design since different uses of IPFS over its life may call for different approaches to privacy. Explicitly implementing an approach to privacy within the IPFS core could "box in" future builders and implementers due to a lack of modularity, flexibility, and future-proofing.

Reviewer 3: "with key principles of protocol design"

"of the protocol design" ? or do you mean protocol design generally? (seems a strong assertion)

Reviewer 3: This paragraph leads with "not built in", but I'd prefer to flip that negative to a positive as you do later in the paragraph.

Author: Rearranged and clarified in https://github.com/ipfs/ipfs-docs/pull/554/commits/d387be8909eae4c3ce2166367505952c741f549d

> By design, IPFS is a public network: Nodes participating in the network store data affiliated with globally consistent [content addresses](/concepts/content-addressing) (CIDs) and advertise that they have those CIDs available for other nodes to use through publicly viewable [distributed hash tables](/concepts/dht/) (DHTs). Indeed, this is one of IPFS's core strengths — at its most basic, it's a sort of globally distributed "server" of the network's total available data, referenceable both by the content itself (those CIDs) and by the participants who have or want the content (the nodes).

Reviewer 3: confusing to have "(the nodes)" after "the content", should move to after "participants"?

Reviewer 3: s/sort of// or something clearer.

Author: https://github.com/ipfs/ipfs-docs/pull/554/commits/6ed4faa9df2826b17d8964b49faf63e2fbabc582 should take care of both of these

This is one of the advantages of IPFS over traditional legacy-web hosting. It means retrieving files — especially popular ones that exist on lots of nodes in the network — can be faster and more bandwidth-efficient. However, it's important to note that those DHT queries happen in public. Because of this, it's possible that third parties could be monitoring this traffic to determine what CIDs are being requested, when, and by whom. As IPFS continues to grow in popularity, it's more likely that such monitoring will exist.

> ### Node IDs and node requests

Reviewer 3: Should we use "peer ids" throughout (as you do below) instead of "node ids"? Or could just changes to "Node identifiability" and leave it at that.

Author: General "Node identifiability" is clearer. Changed in https://github.com/ipfs/ipfs-docs/pull/554/commits/f19e9ba35296fc2cfb12fd861384c735b704774d

The other half of the equation when considering the prospect of IPFS traffic monitoring is that nodes' unique identifiers are themselves public. Just like with CIDs, every individual IPFS node has its own public identifier (known as a PeerID), such as `QmRGgYP1P5bjgapLaShMVhGMSwGN9SfYG3CM2TfhpJ3igE`.

While a long string of letters and numbers may not be a "Johnny Appleseed" level of human-readable specificity, your PeerID is still a long-lived, unique identifier for your node. Keep in mind that it's possible to do a DHT lookup on your PeerID and, particularly if your node is regularly running from the same location (like your home), find your IP address. Additionally, longer-term monitoring of the public IPFS network could yield information about what CIDs your node is requesting and/or reproviding and when.

## Enhancing your IPFS privacy

If there are situations in which you know you'll need to remain private but still want to use IPFS, one of the approaches outlined below may help. And don't forget, you can always discuss privacy and get others' input or ideas in the official [IPFS forums](https://discuss.ipfs.io).

> ### Disabling reproviding

Reviewer 3: This header should be something the uninitiated will know what it means, eg "Controlling what you share" or something like that.

Author: Resolved in https://github.com/ipfs/ipfs-docs/pull/554/commits/435ae665d55a8a59738b9fcef93f7a9fc673b18b

> By default, an IPFS node announces itself as a provider for every CID in its cache. If you'd like to disable this behavior, you can do so in the [reprovider settings](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#reprovider) of your node's config file.

Reviewer 3: Ditto - should define "provider" first, or use something like "announces to the network that it is sharing every CID..." etc

Author: Resolved in https://github.com/ipfs/ipfs-docs/pull/554/commits/61e68df6fbe3d9afc03efa322436058baf887b72

> Disabling default reprovision still enables you to provide data that you've pinned, so it's still possible for you to provide other nodes with content that you care about and want to make sure continues to be available over IPFS.

Reviewer 3: reproviding?

The sentence is confusing after reading the above section. Kinda sounds like "You just turned off the sharing but it still shares". Need some disambiguation between automatic providing vs ... whatever the alternate ways to share are. Explicit pins maybe?

Author: https://github.com/ipfs/ipfs-docs/pull/554/commits/d0f0e3e3149fcea46a3b9931f0e0b7e27d48595e should clarify this.

### Using Tor

If you're familiar with [Tor](https://www.torproject.org/) and comfortable with the command line, you may wish to try [running IPFS over Tor transport](https://dweb-primer.ipfs.io/avenues-for-access/tor-transport) by configuring your node's settings.

If you're a developer building on IPFS, it's worth noting that the global IPFS community continues to experiment with using Tor transport — see [this example from e-commerce organization OpenBazaar](https://github.com/OpenBazaar/go-onion-transport) — and there may already be an open-source codebase to help your own project achieve this.

### Encrypting content transported via IPFS

> If your privacy concerns are less about the potential for monitoring and more about the visibility of the IPFS-provided content itself, this can be mitigated simply by encrypting the content before adding it to the IPFS network. While traffic involving the encrypted content could still be tracked, the _data_ represented by encrypted content's CIDs remains unreadable by anyone without the ability to decrypt it.

Reviewer 3: Need to caveat the safety in encrypting content and putting it on a public network. This is a non-starter for some folks, as future breakthroughs in computing may allow going back and decrypting older public content.

Author: Good point: added in https://github.com/ipfs/ipfs-docs/pull/554/commits/429c780e82193a4511c86ef9b933f9da68473e43

---
title: Privacy
sidebarDepth: 0
description: Learn about user privacy in IPFS and why it does not come with a built-in privacy layer or encryption.
related:
  'Article: IPFS Privacy (Textile)': https://medium.com/pinata/ipfs-privacy-711f4b72b2ea
---

# IPFS and privacy

> As a protocol for peer-to-peer data storage and delivery, IPFS is by design a _public network_: Nodes participating in the network store data affiliated with globally consistent [content addresses](/concepts/content-addressing) (CIDs) and advertise that they have those CIDs available for other nodes to use through publicly viewable [distributed hash tables](/concepts/dht/) (DHTs). This paradigm is one of IPFS's core strengths — at its most basic, it's essentially a globally distributed "server" of the network's total available data, referenceable both by the content itself (those CIDs) and by the participants (the nodes) who have or want the content.

Author: If the text after a colon can stand on its own as a complete sentence, capitalize the first word. If it can't, leave lowercase. So ...

_This much I know to be true: My cat is the softest cat ever._
but
_My cat: so soft._

> What this does mean, however, is that IPFS itself isn't explicitly protecting knowledge _about_ CIDs and the nodes that provide or retrieve them. This isn't something unique to the distributed web; on both the d-web and the legacy web, traffic and other metadata can be monitored in ways that can infer a lot about a network and its users. Some key details on this are outlined below, but in short: While IPFS traffic _between nodes_ is encrypted, the metadata those nodes publish to the DHT is public. Nodes announce a variety of information essential to the DHT's function — including their unique node identifiers (PeerIDs) and the CIDs of data that they're providing — and because of this, information about which nodes are retrieving and/or reproviding which CIDs is publicly available.

Author: I believe @johnnymatthews did this - Johnny, is this house style now?

> What this does mean, however, is that IPFS itself isn't explicitly protecting knowledge _about_ CIDs and the nodes that provide or retrieve them. This isn't something unique to the distributed web; on both the d-web and the legacy web, traffic and other metadata can be monitored in ways that can infer a lot about a network and its users. Some key details on this are outlined below, but in short: While IPFS traffic _between nodes_ is encrypted, the metadata those nodes publish to the DHT is public. Nodes announce a variety of information essential to the DHT's function — including their unique node identifiers (PeerIDs) and the CIDs of data that they're providing — and because of this, information about which nodes are retrieving and/or reproviding which CIDs is publicly available.

Reviewer: Yeah, I thought I'd sneak this in. I'm really not a fan of `dweb` because it reads like _dweeb_. But every time the whole `dweb`, `d-web`, `Dweb` discussion comes up, we never really seem to come to an answer. So I've just been covertly using `d-web`.

So I guess in a sense, `d-web` is the house style. But not by popular vote, or any vote for that matter.

> What this does mean, however, is that IPFS itself isn't explicitly protecting knowledge _about_ CIDs and the nodes that provide or retrieve them. This isn't something unique to the distributed web; on both the d-web and the legacy web, traffic and other metadata can be monitored in ways that can infer a lot about a network and its users. Some key details on this are outlined below, but in short: While IPFS traffic _between nodes_ is encrypted, the metadata those nodes publish to the DHT is public. Nodes announce a variety of information essential to the DHT's function — including their unique node identifiers (PeerIDs) and the CIDs of data that they're providing — and because of this, information about which nodes are retrieving and/or reproviding which CIDs is publicly available.

Reviewer 2: For the record, if this was intentional, I don't feel either way about this, feel free to ignore this and keep as `d-web` :)

> What this does mean, however, is that IPFS itself isn't explicitly protecting knowledge _about_ CIDs and the nodes that provide or retrieve them. This isn't something unique to the distributed web; on both the d-web and the legacy web, traffic and other metadata can be monitored in ways that can infer a lot about a network and its users. Some key details on this are outlined below, but in short: While IPFS traffic _between nodes_ is encrypted, the metadata those nodes publish to the DHT is public. Nodes announce a variety of information essential to the DHT's function — including their unique node identifiers (PeerIDs) and the CIDs of data that they're providing — and because of this, information about which nodes are retrieving and/or reproviding which CIDs is publicly available.

Author: The cynic in me says "LET'S TAKE BACK 'DWEEB'!!1!" but totally happy to defer on this one ;)

If you're worried about the implications of this for your own personal use case, it's worth taking additional measures such as disabling reproviding, encrypting sensitive content, or even running a private IPFS network if that's appropriate for you. More details on these are below.

::: tip
While IPFS traffic _between nodes_ is encrypted, the essential metadata that nodes publish to the DHT — including their unique node identifiers (PeerIDs) and the CIDs of data that they're providing — is public. If you're worried about the implications of this for your personal use case, it's worth taking additional measures.
:::

## What's public on IPFS

All traffic on IPFS is public, including the contents of files themselves (unless they're encrypted; more about this below). For purposes of understanding IPFS privacy, this may be easiest to think about in two halves: content identifiers (CIDs) and IPFS nodes themselves.

### Content identifiers

> Because IPFS uses [content addressing](/concepts/content-addressing/) rather than the legacy web's method of location addressing, each piece of data stored in the IPFS network gets its own unique content identifier (CID). Copies of the data associated with that CID can be stored in any number of locations worldwide on any number of participating IPFS nodes. To make retrieving the data associated with a particular CID efficient and robust, IPFS uses a [distributed hash table](/concepts/dht/) (DHT) to keep track of what's stored where. When you use IPFS to retrieve a particular CID, your node queries the DHT to find the closest nodes to you with that item — and by default also agrees to reprovide that CID to other nodes for a limited time, until periodic "garbage collection" clears your cache of content you haven't used in a while. You can also "pin" CIDs that you want to make sure are never garbage-collected — either explicitly using IPFS's low-level `pin` API, or implicitly using the [Mutable File System](/concepts/file-systems/#mutable-file-system-mfs) (MFS) — which also means you're acting as a permanent reprovider of that data.

Reviewer 4: Is the DHT lookup for all nodes or just DHT servers? Maybe also add the suggestion for how to reset your peerid

Reviewer 4: (Sorry, this should be down on “identifiability” ex your peerid relating to your IP address)

Author: https://github.com/ipfs/ipfs-docs/pull/554/commits/b2e01c5cb2eebde792fb4cc74d2db84b7bc41a0a and https://github.com/ipfs/ipfs-docs/pull/554/commits/0afc26419f7d43fe1f4cd5b7968640df319eba42 add info on resetting PeerID.

### Using Tor

If you're familiar with [Tor](https://www.torproject.org/) and comfortable with the command line, you may wish to try [running IPFS over Tor transport](https://dweb-primer.ipfs.io/avenues-for-access/tor-transport) by configuring your node's settings.

If you're a developer building on IPFS, it's worth noting that the global IPFS community continues to experiment with using Tor transport — see [this example from e-commerce organization OpenBazaar](https://github.com/OpenBazaar/go-onion-transport) — and there may already be an open-source codebase to help your own project achieve this.

### Encrypting content transported via IPFS

If your privacy concerns are less about the potential for monitoring and more about the visibility of the IPFS-provided content itself, this can be mitigated simply by encrypting the content before adding it to the IPFS network. While traffic involving the encrypted content could still be tracked, the _data_ represented by encrypted content's CIDs remains unreadable by anyone without the ability to decrypt it.

> There's one caveat to keep in mind here: While today's encryption might seem bulletproof _today_, it's not guaranteed that it won't be broken at some point in the future. Future breakthroughs in computing might allow going back and decrypting older content that's been put on a public network such as IPFS.

Reviewer 4: If you want to guard against this potential attack vector, using IPFS private networks where IPFS nodes sit behind connection gates that check request ACLs is a potential design direction

Reviewer 4: I think piñata may have a blog post on this?

Author: Thanks! Updated in https://github.com/ipfs/ipfs-docs/pull/554/commits/72908cf6651efafff816df262a8d7b10a62cb656.
